code/controllers/subsystem/job.dm

```diff
@@ -590,14 +601,136 @@ SUBSYSTEM_DEF(job)
 		else //We ran out of spare locker spawns!
 			break
 
+/// Called in jobs subsystem initialize if LOAD_JOBS_FROM_TXT config flag is set reads jobconfig.toml (or if in legacy mode, jobs.txt) to set all of the datum's values to what the server operator wants.
+/datum/controller/subsystem/job/proc/load_jobs_from_config()
+	var/toml_file = "[global.config.directory]/jobconfig.toml"
 
-/datum/controller/subsystem/job/proc/LoadJobs()
-	var/jobstext = file2text("[global.config.directory]/jobs.txt")
-	for(var/datum/job/job as anything in joinable_occupations)
-		var/regex/jobs = new("[job.title]=(-1|\\d+),(-1|\\d+)")
-		jobs.Find(jobstext)
-		job.total_positions = text2num(jobs.group[1])
-		job.spawn_positions = text2num(jobs.group[2])
+	if(!legacy_mode) // this flag is set during the setup of SSconfig, and all warnings were handled there.
+		var/job_config = rustg_read_toml_file(toml_file)
+
+		for(var/datum/job/occupation as anything in joinable_occupations)
+			var/job_title = occupation.title
+			var/job_key = occupation.config_tag
+			if(!job_config[job_key])
+				message_admins(span_notice("[job_title] (with config key [job_key]) is missing from jobconfig.toml! Using codebase defaults.")) // we add both the title and the config key in case they desync over time and someone needs to fix the box for some reason
+				continue
```

Comment:
Again, the job config has to work like an **override**.

If you want to stay subscribed to code defaults, you would do so by not including a job in the job toml file. if you only want to change some values from code default, you would do so by including only configuration options for those settings, and not including the values you don't want to change from default.

Ie, if i only want to change atmos to have a higher account age requirement from `0` days to `6` days, i should be able to just specify the following as the **entirety** of the toml and have this be valid:

```toml
[ATMOSPHERIC_TECHNICIAN]
"Playtime Requirements" = 60
```

Having to put every single job in the config and specify every single value, locking them in, is asking for trouble down the line when we want to be able to change the code defaults and have them take effect on the servers that didn't specify otherwise.

This also means that the entirety of the repo toml and the generated toml has to start commented out (except for anything specified in the legacy config, allowing people to uncomment things they want to change from repo default while serving as a documentation of what repo default is.

Reply:
Okay, I think I understood it now. I'm pretty stupid and lazy (mostly, I couldn't think of a better way to do it), so I just purposefully introduced a key mis-match. Every time we generate jobconfig.toml, we will always write to it with commented out keys. However, if we're importing from jobs.txt and the variables were previously changed there, that will port as uncommented. If we regenerate and the specific line was uncommented, the behavior should allow that key to be uncommented in the downloadable file.

That's how it should work in theory. I'm not a very good programmer and I wanted to avoid bugs, so I probably did a bit more overengineering on this system that I needed to. I believe I got what you were saying here, though- and great apologies if I did not because I've had other things take up my time recently and I typically have to relearn what I was doing before working on the system again.

A briefer way of putting it: I'm not purposefully trying to irritate anyone. Thank you for your patience thus far.